File: test_gemini.js
fetch('https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash:generateContent?key=AIzaSyDSZkTlHvFB7DKUldl9ArAmaTrYknwLBWk', {
  method: 'POST',
  headers: { 'Content-Type': 'application/json' },
  body: JSON.stringify({ contents: [{ parts: [{ text: 'hi' }] }] })
}).then(r => r.text()).then(console.log).catch(console.error);

File: test_gemini.py
import urllib.request, json
url = "https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash:generateContent?key=AIzaSyDSZkTlHvFB7DKUldl9ArAmaTrYknwLBWk"
req = urllib.request.Request(url, json.dumps({"contents":[{"parts":[{"text":"hi"}]}]}).encode('utf-8'), {'Content-Type': 'application/json'})
try:
    print(urllib.request.urlopen(req).read().decode())
except Exception as e:
    print(e.read().decode())
